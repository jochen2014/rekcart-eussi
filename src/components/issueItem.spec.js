import React from 'react';
import renderer from 'react-test-renderer';
import IssueItem from './issueItem';

describe('issue item', () => {
    it('should render correctly', () => {
        Date.now = jest.genMockFunction().mockReturnValue(1507342218559); // 7/10/2017 13:10
        const data = {
            "id": 111,
            "number": 1,
            "state": "open",
            "title": "unit test",
            "user": {
                "id": 1,
                "avatarUrl": "https://avatars3.githubusercontent.com/u/1?v=4",
                "name": "bochen"
            },
            "dateCreated_raw": "2017-10-05T10:55:42Z",
            "dateCraetedStr": "14 minutes ago",
            "tags": ["CLA Signed"]
        }
        const tree = renderer.create(<IssueItem issue={data} />).toJSON();
        expect(tree).toMatchSnapshot();
    })
})