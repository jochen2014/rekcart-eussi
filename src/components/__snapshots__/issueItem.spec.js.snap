// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`issue item should render correctly 1`] = `
<li
  className="issue-item"
>
  <div
    className="header"
  >
    <a
      className="link title"
      href="#/issues/111"
    >
      unit test
    </a>
    <span
      className="tag-0"
    >
      CLA Signed
    </span>
  </div>
  <div
    className="content"
  >
    <div>
      #
      1
    </div>
    <img
      className="avatar"
      src="https://avatars3.githubusercontent.com/u/1?v=4"
    />
    <div>
      <a
        className="link"
        href="#/users/1"
      >
        bochen
      </a>
      opened 
      14 minutes ago
    </div>
  </div>
</li>
`;
